targets/csharp-unityv2/source/Shared/Internal/PlayFabHttp/PlayFabWebRequest.cs

```diff
@@ -0,0 +1,596 @@
+                if (cloudScriptUrl != null)
+                {
+                    PlayFabSettings.LogicServerUrl = cloudScriptUrl.Url;
+                }
+#endif
+                //Send our post event.  //TODO: maybe we want this Enqueued also?
+                PlayFabHttp.SendEvent(request, result, ApiProcessingEventType.Post);
+
+                if (callBack != null)
+                {
+                    lock (ResultQueue)
+                    {
+                        //Queue The result callbacks to run on the main thread.
+                        ResultQueue.Enqueue(() => {
+#if PLAYFAB_REQUEST_TIMING
+                            state.Stopwatch.Stop();
+                            state.Timing.MainThreadRequestMs = (int)state.Stopwatch.ElapsedMilliseconds;
+                            PlayFabHttp.SendRequestTiming(state.Timing);
+#endif
+                            callBack(result);
+                        });
+                    }
+                }
+
+            }
+            catch (Exception e)
+            {
+                Debug.LogException(e);
+                state.State = HttpRequestState.Error;
+            }
+
+        }
+
+        
+        public void Update()
+        {
+            lock (ResultQueue)
+            {
+                while (ResultQueue.Count > 0)
+                {
+                    var actionToQueue = ResultQueue.Dequeue();
+                    _tempActions.Enqueue(actionToQueue);
+                }
+            }
+
+            while (_tempActions.Count > 0)
+            {
+                var finishedRequest = _tempActions.Dequeue();
+                finishedRequest.Invoke();
+            }
+        }
+
+        public static string ResponseToString(HttpWebResponse webResponse)
+        {
+            try
+            {
+                using (var responseStream = webResponse.GetResponseStream())
+                {
+                    if (responseStream == null)
+                        return null;
+                    using (var stream = new System.IO.StreamReader(responseStream))
+                    {
+                        return stream.ReadToEnd();
+                    }
+                }
+
+            }
+            catch (WebException e)
+            {
+                Debug.LogException(e);
+                return null;
+            }
+            catch (Exception e)
+            {
+                Debug.LogException(e);
+                return null;
+            }
+        }
+
+        public int GetPendingMessages()
+        {
+            int count = 0;
+            lock (ActiveRequests)
+                count += ActiveRequests.Count + _activeCallCount;
+            lock (ResultQueue)
+                count += ResultQueue.Count;
+            return count;
+        }
+
+        public enum HttpRequestState
+        {
+            Sent,
+            Received,
+            Idle,
+            Error
+        }
+
+        public class RequestState
```

Comment:
This isn't a state.
HttpRequestState State; is the state.
This is a container.  A "CallRequestContainer" if you will...
And
// This class stores all the data relevant to a a single API call.